### PR TITLE
Update battle messages to Gen 5+ standards

### PR DESCRIPTION
**Pull request by kittenchilly**
## Description
Updates almost every vanilla battle message to be consistent with gen 5+ and a lot of the custom messages we add. This results in better grammar and decapitalization of terms in most cases.
Also changes a few scripts that pertain to battle messages, as well as adding new weather and terrain strings if caused due to an overworld condition.

## Images
pokeemerald-0

## **Discord contact info**
kittenchilly

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Before you continue modifying this, please fix failing test messages.

File: src/battle_message.c [hunk 1]
```diff
@@ -56,264 +56,264 @@ EWRAM_DATA struct BattleMsgData *gBattleMsgDataPtr = NULL;
...
+static const u8 sText_PkmnEnduredHit[] = _("{B_DEF_NAME_WITH_PREFIX} endured\nthe hit!");
+static const u8 sText_MagnitudeStrength[] = _("Magnitude {B_BUFF1}!");
+static const u8 sText_PkmnCutHPMaxedAttack[] = _("{B_ATK_NAME_WITH_PREFIX} cut its own HP\nand maximized its Attack!");
 static const u8 sText_PkmnCopiedStatChanges[] = _("{B_ATK_NAME_WITH_PREFIX} copied\n{B_DEF_NAME_WITH_PREFIX}'s stat changes!");
-static const u8 sText_PkmnGotFree[] = _("{B_ATK_NAME_WITH_PREFIX} got free of\n{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}!");
-static const u8 sText_PkmnShedLeechSeed[] = _("{B_ATK_NAME_WITH_PREFIX} shed\nLEECH SEED!");
-static const u8 sText_PkmnBlewAwaySpikes[] = _("{B_ATK_NAME_WITH_PREFIX} blew away\nspikes!");
+static const u8 sText_PkmnGotFree[] = _("{B_ATK_NAME_WITH_PREFIX} got free of\n{B_DEF_NAME_WITH_PREFIX}'s {B_BUFF1}!"); //not in gen 5+, generic rapid spin?
+static const u8 sText_PkmnShedLeechSeed[] = _("{B_ATK_NAME_WITH_PREFIX} shed\nLeech Seed!"); //not in gen 5+, generic rapid spin?
+static const u8 sText_PkmnBlewAwaySpikes[] = _("{B_ATK_NAME_WITH_PREFIX} blew away\nspikes!"); //not in gen 5+, generic rapid spin?
```
Comment (MandL27) [inline src/battle_message.c:228]: Gens 5+ use generic entry hazard removal text for this; unsure on Leech Seed tho

File: src/battle_message.c [hunk 2]
```diff
@@ -327,66 +327,66 @@ static const u8 sText_StatsWontIncrease2[] = _("{B_ATK_NAME_WITH_PREFIX}'s stats
...
 static const u8 sText_SuperEffective[] = _("It's super effective!");
-static const u8 sText_GotAwaySafely[] = _("{PLAY_SE SE_FLEE}Got away safely!\p");
+static const u8 sText_GotAwaySafely[] = _("{PLAY_SE SE_FLEE}You got away safely!!\p");
 static const u8 sText_PkmnFledUsingIts[] = _("{PLAY_SE SE_FLEE}{B_ATK_NAME_WITH_PREFIX} fled\nusing its {B_LAST_ITEM}!\p");
 static const u8 sText_PkmnFledUsing[] = _("{PLAY_SE SE_FLEE}{B_ATK_NAME_WITH_PREFIX} fled\nusing {B_ATK_ABILITY}!\p");
-static const u8 sText_WildPkmnFled[] = _("{PLAY_SE SE_FLEE}Wild {B_BUFF1} fled!");
+static const u8 sText_WildPkmnFled[] = _("{PLAY_SE SE_FLEE}The wild {B_BUFF1} fled!");
 static const u8 sText_PlayerDefeatedLinkTrainer[] = _("Player defeated\n{B_LINK_OPPONENT1_NAME}!");
-static const u8 sText_TwoLinkTrainersDefeated[] = _("Player beat {B_LINK_OPPONENT1_NAME}\nand {B_LINK_OPPONENT2_NAME}!");
+static const u8 sText_TwoLinkTrainersDefeated[] = _("Player defeated {B_LINK_OPPONENT1_NAME}\nand {B_LINK_OPPONENT2_NAME}!");
```
Comment (MandL27) [inline src/battle_message.c:340]: At least in gen 9, this starts with `You defeated` instead (same for other victory strings)

File: src/battle_message.c [hunk 3]
```diff
@@ -327,66 +327,66 @@ static const u8 sText_StatsWontIncrease2[] = _("{B_ATK_NAME_WITH_PREFIX}'s stats
...
 static const u8 sText_RainStopped[] = _("The rain stopped.");
-static const u8 sText_SandstormBrewed[] = _("A sandstorm brewed!");
-static const u8 sText_SandstormRages[] = _("The sandstorm rages.");
+static const u8 sText_SandstormBrewed[] = _("A sandstorm kicked up!");
+static const u8 sText_SandstormRages[] = _("The sandstorm rages."); //not in gen 5+
 static const u8 sText_SandstormSubsided[] = _("The sandstorm subsided.");
-static const u8 sText_SunlightGotBright[] = _("The sunlight got bright!");
-static const u8 sText_SunlightStrong[] = _("The sunlight is strong.");
+static const u8 sText_SunlightGotBright[] = _("The sunlight turned harsh!");
+static const u8 sText_SunlightStrong[] = _("The sunlight is strong."); //not in gen 5+
```
Comment (MandL27) [inline src/battle_message.c:363]: Gen 8 uses `The sunlight is harsh!` for overworld sun; maybe use that as precedent for a reword?

File: src/battle_message.c [hunk 4]
```diff
@@ -474,45 +475,45 @@ const u8 *const gPokeblockWasTooXStringTable[FLAVOR_COUNT] =
...
-static const u8 sText_GiveNicknameCaptured[] = _("Give a nickname to the\ncaptured {B_DEF_NAME}?");
-static const u8 sText_PkmnSentToPC[] = _("{B_DEF_NAME} was sent to\n{B_PC_CREATOR_NAME} PC.");
+static const u8 sText_GiveNicknameCaptured[] = _("Give a nickname to the\ncaught {B_DEF_NAME}?");
+static const u8 sText_PkmnSentToPC[] = _("{B_DEF_NAME} was sent to\n{B_PC_CREATOR_NAME} PC."); //not updating this since the terminology is different
 static const u8 sText_Someones[] = _("someone's");
-static const u8 sText_Lanettes[] = _("LANETTE's");
-static const u8 sText_PkmnDataAddedToDex[] = _("{B_DEF_NAME}'s data was\nadded to the POKéDEX.\p");
+static const u8 sText_Lanettes[] = _("LANETTE's"); //no decapitalize until it is everywhere
+static const u8 sText_PkmnDataAddedToDex[] = _("{B_DEF_NAME}'s data was\nadded to the Pokédex.\p");
 static const u8 sText_ItIsRaining[] = _("It is raining.");
```
Comment (MandL27) [inline src/battle_message.c:495]: Became `It's raining!` in gen 8

File: src/battle_message.c [hunk 5]
```diff
@@ -474,45 +475,45 @@ const u8 *const gPokeblockWasTooXStringTable[FLAVOR_COUNT] =
...
 static const u8 sText_SandstormIsRaging[] = _("A sandstorm is raging.");
```
Comment (MandL27) [inline src/battle_message.c:496]: Became `The sandstorm is raging!` in gen 8

**Review by MandL27 — COMMENTED** [review]
Looks mostly good to me. Saw a few strings that could use further adjustment. Didn't review tests yet.

File: src/battle_message.c [hunk 6]
```diff
@@ -1986,8 +1987,8 @@ const u8 gText_LineBreak[] = _("\l");
 const u8 gText_NewLine[] = _("\n");
 const u8 gText_Are[] = _("are");
 const u8 gText_Are2[] = _("are");
-const u8 gText_BadEgg[] = _("Bad EGG");
-const u8 gText_BattleWallyName[] = _("WALLY");
+const u8 gText_BadEgg[] = _("Bad Egg");
+const u8 gText_BattleWallyName[] = _("WALLY");
```
Comment (MandL27) [inline src/battle_message.c:1991]: Missed this one in the review - was a trailing space added here?

File: src/battle_message.c (hunk 3 above)
Comment (kittenchilly) [reply src/battle_message.c:363]: I'm not going to change it since it'd be different from the other ones if so.

Comment (kittenchilly) [reply src/battle_message.c:363]: Nevermind, I'm going to just add a new string for it.

**Comment by kittenchilly** [comment]
Finally got all tests to pass. Should be reviewable now

**Comment by AsparagusEduardo** [comment]
Bruh.
image

File: src/battle_message.c [hunk 7]
```diff
@@ -57,264 +57,264 @@ EWRAM_DATA struct BattleMsgData *gBattleMsgDataPtr = NULL;
...
-static const u8 sText_PkmnTookFoe[] = _("{B_DEF_NAME_WITH_PREFIX} took\n{B_ATK_NAME_WITH_PREFIX} with it!");
-static const u8 sText_PkmnReducedPP[] = _("Reduced {B_DEF_NAME_WITH_PREFIX}'s\n{B_BUFF1} by {B_BUFF2}!");
+static const u8 sText_PkmnMoveDisabledNoMore[] = _("{B_ATK_NAME_WITH_PREFIX} is no\nlonger disabled!");
+static const u8 sText_PkmnGotEncore[] = _("{B_DEF_NAME_WITH_PREFIX} received\nan encore!");
+static const u8 sText_PkmnEncoreEnded[] = _("{B_ATK_NAME_WITH_PREFIX}'s encore\nended!");
+static const u8 sText_PkmnTookAim[] = _("{B_ATK_NAME_WITH_PREFIX} took aim at\n{B_DEF_NAME_WITH_PREFIX}!");
+static const u8 sText_PkmnSketchedMove[] = _("{B_ATK_NAME_WITH_PREFIX} sketched\n{B_BUFF1}!");
+static const u8 sText_PkmnTryingToTakeFoe[] = _("{B_ATK_NAME_WITH_PREFIX} is trying\nto take its foe down with it!");
+static const u8 sText_PkmnTookFoe[] = _("{B_DEF_NAME_WITH_PREFIX} took its attacker\ndown with it!");
+static const u8 sText_PkmnReducedPP[] = _("It reduced the PP of {B_DEF_NAME_WITH_PREFIX}'s\n{B_BUFF1} by {B_BUFF2}!");
```
Comment (MandL27) [inline src/battle_message.c:211]: Does this fit with where the linebreak is? I feel like that'd easily overflow if there's both a prefix and a medium-long mon name

File: src/battle_message.c [hunk 8]
```diff
@@ -57,264 +57,264 @@ EWRAM_DATA struct BattleMsgData *gBattleMsgDataPtr = NULL;
...
-static const u8 sText_PkmnSwappedAbilities[] = _("{B_ATK_NAME_WITH_PREFIX} swapped abilities\nwith its opponent!");
-static const u8 sText_PkmnSealedOpponentMove[] = _("{B_ATK_NAME_WITH_PREFIX} sealed the\nopponent's move(s)!");
-static const u8 sText_PkmnWantsGrudge[] = _("{B_ATK_NAME_WITH_PREFIX} wants the\nopponent to bear a GRUDGE!");
-static const u8 sText_PkmnLostPPGrudge[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_BUFF1} lost\nall its PP due to the GRUDGE!");
-static const u8 sText_PkmnShroudedItself[] = _("{B_ATK_NAME_WITH_PREFIX} shrouded\nitself in {B_CURRENT_MOVE}!");
-static const u8 sText_PkmnMoveBounced[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_CURRENT_MOVE}\nwas bounced back by MAGIC COAT!");
+static const u8 sText_PkmnSwappedAbilities[] = _("{B_ATK_NAME_WITH_PREFIX} swapped\nAbilities with its target!");
+static const u8 sText_PkmnSealedOpponentMove[] = _("{B_ATK_NAME_WITH_PREFIX} sealed any moves\nits target shares with it!");
+static const u8 sText_PkmnWantsGrudge[] = _("{B_ATK_NAME_WITH_PREFIX} wants its target\nto bear a grudge!");
+static const u8 sText_PkmnLostPPGrudge[] = _("{B_ATK_NAME_WITH_PREFIX}'s {B_BUFF1} lost all its PP\ndue to the grudge!");
```
Comment (MandL27) [inline src/battle_message.c:258]: Another line length concern

File: src/battle_message.c [hunk 9]
```diff
@@ -57,264 +57,264 @@ EWRAM_DATA struct BattleMsgData *gBattleMsgDataPtr = NULL;
...
+static const u8 sText_PkmnsXWhippedUpSandstorm[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nwhipped up a sandstorm!"); //not in gen 5+, ability popup
+static const u8 sText_PkmnsXIntensifiedSun[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nintensified the sun's rays!"); //not in gen 5+, ability popup
+static const u8 sText_PkmnsXPreventsYLoss[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\nprevents {B_BUFF1} loss!"); //not in gen 5+, ability popup
+static const u8 sText_PkmnsXInfatuatedY[] = _("{B_DEF_NAME_WITH_PREFIX}'s {B_DEF_ABILITY}\ninfatuated {B_ATK_NAME_WITH_PREFIX}!"); //not in gen 5+, ability popup
+static const u8 sText_PkmnsXMadeYIneffective[] = _("{B_DEF_NAME_WITH_PREFIX}'s {B_DEF_ABILITY}\nmade {B_CURRENT_MOVE} ineffective!"); //not in gen 5+, ability popup
+static const u8 sText_PkmnsXCuredYProblem[] = _("{B_SCR_ACTIVE_NAME_WITH_PREFIX}'s {B_SCR_ACTIVE_ABILITY}\ncured its {B_BUFF1} problem!"); //not in gen 5+, ability popup
+static const u8 sText_ItSuckedLiquidOoze[] = _("{B_DEF_NAME_WITH_PREFIX} sucked up the\nliquid ooze!");
```
Comment (MandL27) [inline src/battle_message.c:307]: Nitpick, but the line break here feels super lopsided

File: src/battle_message.c [hunk 10]
```diff
@@ -2112,9 +2132,9 @@ const u8 gText_Body[] = _("Body");
 const u8 gText_Judgment[] = _("{B_BUFF1}{CLEAR 13}Judgment{CLEAR 13}{B_BUFF2}");
 static const u8 sText_TwoTrainersSentPkmn[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} sent\nout {B_OPPONENT_MON1_NAME}!\p{B_TRAINER2_CLASS} {B_TRAINER2_NAME} sent\nout {B_OPPONENT_MON2_NAME}!");
 static const u8 sText_Trainer2SentOutPkmn[] = _("{B_TRAINER2_CLASS} {B_TRAINER2_NAME} sent\nout {B_BUFF1}!");
-static const u8 sText_TwoTrainersWantToBattle[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} and\n{B_TRAINER2_CLASS} {B_TRAINER2_NAME}\lwant to battle!\p");
+static const u8 sText_TwoTrainersWantToBattle[] = _("You are challenged by\n{B_TRAINER1_CLASS} {B_TRAINER1_NAME} and\l{B_TRAINER2_CLASS} {B_TRAINER2_NAME}!\p");
 static const u8 sText_InGamePartnerSentOutZGoN[] = _("{B_PARTNER_CLASS} {B_PARTNER_NAME} sent\nout {B_PLAYER_MON2_NAME}!\lGo, {B_PLAYER_MON1_NAME}!");
-static const u8 sText_TwoInGameTrainersDefeated[] = _("{B_TRAINER1_CLASS} {B_TRAINER1_NAME} and\n{B_TRAINER2_CLASS} {B_TRAINER2_NAME}\lwere defeated!\p");
+static const u8 sText_TwoInGameTrainersDefeated[] = _("Player beat\n{B_TRAINER1_CLASS} {B_TRAINER1_NAME}\land {B_TRAINER2_CLASS} {B_TRAINER2_NAME}!\p");
```
Comment (MandL27) [inline src/battle_message.c:2137]: Pretty sure this should be "You defeated [...]"

**Review by MandL27 — CHANGES_REQUESTED** [review]
Going back through and reviewing this more thoroughly now. Spotted a couple possible line length issues and a missed terminology update.

Nitpick and possibly out of scope, but should renamed trainer classes and grammar involving trainer classes be included in this as well?

**Comment by AlexOn1ine** [comment]
Conflicts

File: charmap.txt [hunk 11]
```diff
@@ -412,6 +412,10 @@ B_DEF_TEAM1 = FD 3A
 B_DEF_TEAM2 = FD 3B
 @ FD 3C - preiously gActiveBattler
 @ FD 3D - preiously gActiveBattler without Illusion Check
+B_ATK_NAME_WITH_PREFIX2 = FD 3E
+B_DEF_NAME_WITH_PREFIX2 = FD 3F
+B_EFF_NAME_WITH_PREFIX2 = FD 40
+B_SCR_ACTIVE_NAME_WITH_PREFIX2 = FD 41
```
Comment (AlexOn1ine) [inline charmap.txt:418]: what's the reason for those?

Comment (AlexOn1ine) [reply charmap.txt:418]: I understand, all good

Comment (kittenchilly) [reply charmap.txt:418]: Yes, because of capitalization of the "the"

File: src/battle_message.c [hunk 12]
```diff
@@ -545,106 +556,106 @@ static const u8 sText_TargetAbilityActivates[] = _("{B_DEF_NAME_WITH_PREFIX}'s {
...
 static const u8 sText_PkmnSwitchedAtkAndDef[] = _("{B_ATK_NAME_WITH_PREFIX} switched its\nAttack and Defense!");
-static const u8 sText_PkmnsAbilitySuppressed[] = _("{B_DEF_NAME_WITH_PREFIX}'s ability\nwas suppressed!");
-static const u8 sText_ShieldedFromCriticalHits[] = _("The {B_CURRENT_MOVE} shielded {B_ATK_TEAM2}\nteam from critical hits!");
-static const u8 sText_SwitchedAtkAndSpAtk[] = _("{B_ATK_NAME_WITH_PREFIX} switched all its\nchanges to its Attack and\pSp. Atk with the target!");
-static const u8 sText_SwitchedDefAndSpDef[] = _("{B_ATK_NAME_WITH_PREFIX} switched all its\nchanges to its Defense and\pSp. Def with the target!");
+static const u8 sText_PkmnsAbilitySuppressed[] = _("{B_DEF_NAME_WITH_PREFIX}'s Ability\nwas suppressed!");
+static const u8 sText_ShieldedFromCriticalHits[] = _("Lucky Chant shielded {B_ATK_TEAM2} team from critical hits!");
+static const u8 sText_SwitchedAtkAndSpAtk[] = _("{B_ATK_NAME_WITH_PREFIX} switched all changes\nto its Attack and Sp. Atk\pwith its target!");
+static const u8 sText_SwitchedDefAndSpDef[] = _("{B_ATK_NAME_WITH_PREFIX} switched all changes\nto its Defense andSp. Def\pwith its target!");
```
Comment (Aeliko) [inline src/battle_message.c:569]: Missing space

**Comment by AsparagusEduardo** [comment]
From what I looked at when making the PR to it, so far looks good. Once we get the remaining text fixed and get the text test ready, should be good to merge.

File: src/battle_message.c [hunk 13]
```diff
@@ -1551,7 +1567,7 @@ const u8 *const gBattleStringsTable[BATTLESTRINGS_COUNT] =
     [STRINGID_PKMNFROSTBITEHEALED - BATTLESTRINGS_TABLE_START] = sText_PkmnFrostbiteHealed,
     [STRINGID_PKMNFROSTBITEHEALED2 - BATTLESTRINGS_TABLE_START] = sText_PkmnFrostbiteHealed2,
     [STRINGID_PKMNFROSTBITEHEALEDBY - BATTLESTRINGS_TABLE_START] = sText_PkmnFrostbiteHealedBy,
-    [STRINGID_ULTRABURSTREACTING - BATTLESTRINGS_TABLE_START] = sText_UltraBurstReacting,
+	[STRINGID_ULTRABURSTREACTING - BATTLESTRINGS_TABLE_START] = sText_UltraBurstReacting,
```
Comment (AsparagusEduardo) [inline src/battle_message.c:1570]:
```suggestion
    [STRINGID_ULTRABURSTREACTING - BATTLESTRINGS_TABLE_START] = sText_UltraBurstReacting,
```

File: src/battle_message.c [hunk 14]
```diff
@@ -1561,6 +1577,14 @@ const u8 *const gBattleStringsTable[BATTLESTRINGS_COUNT] =
     [STRINGID_POKEFLUTECATCHY - BATTLESTRINGS_TABLE_START] = sText_PlayedFluteCatchyTune,
     [STRINGID_POKEFLUTE - BATTLESTRINGS_TABLE_START] = sText_PlayedThe,
     [STRINGID_MONHEARINGFLUTEAWOKE - BATTLESTRINGS_TABLE_START] = sText_PkmnHearingFluteAwoke,
+	[STRINGID_SUNLIGHTISHARSH - BATTLESTRINGS_TABLE_START] = sText_SunlightIsHarsh,
```
Comment (AsparagusEduardo) [inline src/battle_message.c:1580]:
```suggestion
    [STRINGID_SUNLIGHTISHARSH - BATTLESTRINGS_TABLE_START] = sText_SunlightIsHarsh,
```

File: src/battle_message.c [hunk 15]
```diff
@@ -3268,22 +3325,40 @@ static const u8 *BattleStringGetOpponentClassByTrainerId(u16 trainerId)
...
+    // Clear destination first
+    while (dstID < dstSize)
+    {
+        dst[dstID] = EOS;
+        dstID++;
+    }
+
+    dstID = 0;
+    //DebugPrintf("BEGIN:    src:\"%S\"", src);
     while (*src != EOS)
```
Comment (AsparagusEduardo) [inline src/battle_message.c:3356]:
```suggestion
    dstID = 0;
    while (*src != EOS)
```

File: src/battle_message.c [hunk 16]
```diff
@@ -3268,22 +3325,40 @@ static const u8 *BattleStringGetOpponentClassByTrainerId(u16 trainerId)
...
     {
         toCpy = NULL;
+        dstWidth = GetStringLineWidth(fontId, dst, letterSpacing, lineNum, dstSize, FALSE);
+        //DebugPrintf("    line:%d, dstWidth:%d, lastSkip:%d, \"%S\"", lineNum, dstWidth, lastValidSkip, dst);
```
Comment (AsparagusEduardo) [inline src/battle_message.c:3360]:
```suggestion
        dstWidth = GetStringLineWidth(fontId, dst, letterSpacing, lineNum, dstSize, FALSE);
```

File: src/battle_message.c [hunk 17]
```diff
@@ -3658,10 +3733,30 @@ u32 BattleStringExpandPlaceholders(const u8 *src, u8 *dst)
...
+                break;
+            case B_TXT_SCR_ACTIVE_NAME_WITH_PREFIX2:
+                HANDLE_NICKNAME_STRING_LOWERCASE(gBattleScripting.battler)
+                break;
             }
 
             if (toCpy != NULL)
             {
+                toCpyWidth = GetStringLineWidth(fontId, toCpy, letterSpacing, 1, dstSize, FALSE);
+                //DebugPrintf("  toCpy width:%d, \"%S\"", toCpyWidth, toCpy);
```
Comment (AsparagusEduardo) [inline src/battle_message.c:3753]:
```suggestion
                toCpyWidth = GetStringLineWidth(fontId, toCpy, letterSpacing, 1, dstSize, FALSE);
```

**Review by AsparagusEduardo — CHANGES_REQUESTED** [review]
Some of these is stuff I forgot to clean up.

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**